### DISCUSSION TITLE
Network Menu Page Not Loading

### DISCUSSION THREAD

**Discussion by R101G** [rern/rAudio#1811] [Fixed Issues]
The Network Menu option does not load the page.

**Comment by somax80** [comment]
Same problem for me...
Maybe it's related to LAN connection? I hadn't any problem while i was using lan connection, then i've configured wifi connection and reboot unplugging the lan cable ... and ... the network menu option doesn't load anymore.

rAudio 20241108
6.6.45-2-rpi aarch64
Raspberry Pi 3B Rev 1.2

**Reply by R101G** [reply]
Yes, you are right. My Pi3 with LAN works, Pi4 with wifi doesn't. Same rAudio update in both.

**Comment by markblinkhorn** [comment]
I get this on a Pi3+
I found the following in the journal when I try to open the networks page:

```shell
Nov 28 20:25:49 rAudio4 nginx[433]: 2024/11/28 20:25:49 [error] 433#433: *7 FastCGI sent in stderr: "PHP message: PHP Warning:  Undefined variable $menus in /srv/http/settings/function.php on line 158; PHP message: PHP Warning:  foreach() argument must be of type array|object, null given in /srv/http/settings/function.php on line 158" while reading response header from upstream, client: 192.168.160.124, server: localhost, request: "GET /settings.php?p=networks HTTP/1.1", upstream: "fastcgi://unix:/run/php-fpm/php-fpm.sock:", host: "raudio4.local", referrer: "http://raudio4.local/settings.php?p=system"
``` 

Maybe a clue here...

**Reply by markblinkhorn** [reply]
UPDATE: Not such a good clue though - this message is also logged on another Pi3+ that *does* work. Both are WiFi connected. I haven't spotted the difference between them - yet.

**Comment by R101G** [comment]
My Pi4 with wifi is now working OK.

Edit: Bluetooth is not enabled, nor is AP.

**Comment by markblinkhorn** [comment]
Don't you just hate it when...

Mark Blinkhorn
________________________________

**Comment by markblinkhorn** [comment]
So, I copied /srv/http/bash/networks-data.sh from the working machine to the non-working machine. A simple refresh of the Networks page now displays as expected. I don't know which is the 'correct' version of the shell script but the 'non-working' is 3229 bytes and the 'working' is  3216 byte. And, why are they different anyway? Both machines are:

rAudio 20241108
6.12.1-1-rpi aarch64
Raspberry Pi 3B+ Rev 1.4
BCM2837B0 • 904 MiB
4 x Cortex-A53 @ 1.40 GHz

UPDATE: the 'working' version is the older one, not the current one.
And now I get it: i updated rAudio3 (the 'working' one) on 20241109 and rAudio4 (the 'non-working' one) on 20241111. @rern updated networks-data.sh on 20241110 ! - What are the odds? Anyway, if you manually run the current version it displays:
```shell
rAudio4:/srv/http/bash/settings  #  bash networks-data.sh.sav
Too few arguments
{
  "page" : "networks"
, "devicebt"    : true
, "deviceeth"   : true
, "devicewl"    : true
, "ap"          : false
, "apconf"      : false
, "apstartup"   : false
, "camilladsp"  : false
, "connectedwl" : true
, "gateway"     : "192.168.160.254"
, "hostname"    : ""
, "ip"          : ""
, "ipsub"       : "."
, "listbt"      : false
, "listeth"     : false
, "listwl"      : [ {
  "dbm"     : -62
, "gateway" : "192.168.160.254"
, "ip"      : "192.168.160.139"
, "ssid"    : "SI3"
} ]
}
``` 
Whereas the earlier version displays:
```shell
rAudio4:/srv/http/bash/settings  #  bash networks-data.sh
{
  "page" : "networks"
, "devicebt"    : true
, "deviceeth"   : true
, "devicewl"    : true
, "ap"          : false
, "apconf"      : false
, "apstartup"   : false
, "camilladsp"  : false
, "connectedwl" : true
, "gateway"     : "192.168.160.254"
, "hostname"    : "rAudio4.local"
, "ip"          : "192.168.160.139"
, "ipsub"       : "192.168.160."
, "listbt"      : false
, "listeth"     : false
, "listwl"      : [ {
  "dbm"     : -62
, "gateway" : "192.168.160.254"
, "ip"      : "192.168.160.139"
, "ssid"    : "SI3"
} ]
}
```

Hope this helps...

Mark

**Comment by markblinkhorn** [comment]
20241129 fixes it. Thanks @rern
